app/src/main/cpp/skyline/services/codec/IHardwareOpusDecoder.h

```diff
@@ -0,0 +1,150 @@
+// SPDX-License-Identifier: MPL-2.0
+// Copyright © 2021 Skyline Team and Contributors (https://github.com/skyline-emu/)
+
+#pragma once
+
+#include <opus.h>
+
+#include <common.h>
+#include <services/serviceman.h>
+
+namespace skyline::service::codec {
+    /**
+     * @brief This structure is located at the beginning of OpusDataIn for DecodeInterleaved*, with the actual opus packet following this
+     * @note These fields are big-endian
+     * @url https://github.com/switchbrew/libnx/blob/c5a9a909a91657a9818a3b7e18c9b91ff0cbb6e3/nx/include/switch/services/hwopus.h#L19
+     */
+    struct OpusDataHeader {
+        u32 size; //!< Size of the packet following this header
+        u32 finalRange; //!< Indicates the final range of the codec encoder's entropy coder, this can be left at zero
+
+        u32 GetPacketSize() {
+            return util::SwapEndianness(size);
+        }
+    };
+    static_assert(sizeof(OpusDataHeader) == 0x8);
+
+    size_t CalculateOutBufferSize(i32 sampleRate, i32 channelCount, i32 frameSize);
```

Comment:
Why is this not inside the class (and why doesn't it have a comment)? If you need to use it in the constructor then make it a `static` member (preferably `private` as well).

Reply:
This is not inside the class and is not static because it is also used in `IHardwareOpusDecoderManager.cpp`, bylaws also asked me about this in some CR above. I'm open to any suggestions if there's a better way to do this, I chose to put that in the `codec` namespace.

Reply:
I will add a comment though, missed it.